pkg/instrumentation/nethttp/client/client_hook.go

```diff
@@ -0,0 +1,153 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/inst"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/nethttp/semconv"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/shared"
+)
+
+const (
+	otelExporterPrefix     = "OTel OTLP Exporter Go"
+	instrumentationName    = "github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/nethttp"
+	instrumentationVersion = "0.1.0"
+)
+
+var (
+	logger     = shared.GetLogger()
+	tracer     trace.Tracer
+	propagator propagation.TextMapPropagator
+	initOnce   sync.Once
+)
+
+func initInstrumentation() {
+	initOnce.Do(func() {
+		if err := shared.SetupOTelSDK(); err != nil {
+			logger.Error("failed to setup OTel SDK", "error", err)
+		}
+		tracer = otel.GetTracerProvider().Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion(instrumentationVersion),
+		)
+		propagator = otel.GetTextMapPropagator()
+		logger.Info("HTTP client instrumentation initialized")
+	})
+}
+
+// netHttpClientEnabler controls whether client instrumentation is enabled
+type netHttpClientEnabler struct{}
+
+func (n netHttpClientEnabler) Enable() bool {
+	return shared.Instrumented("NETHTTP")
+}
+
+var clientEnabler = netHttpClientEnabler{}
+
+func BeforeRoundTrip(ictx inst.HookContext, transport *http.Transport, req *http.Request) {
+	if !clientEnabler.Enable() {
+		logger.Debug("HTTP client instrumentation disabled")
+		return
+	}
+
+	// Filter out OTel exporter requests to prevent infinite loops
+	if strings.HasPrefix(req.Header.Get("User-Agent"), otelExporterPrefix) {
+		logger.Debug("Skipping OTel exporter request", "user_agent", req.Header.Get("User-Agent"))
+		return
+	}
+
+	initInstrumentation()
+
+	logger.Debug("BeforeRoundTrip called",
+		"method", req.Method,
+		"url", req.URL.String(),
+		"host", req.Host)
+
+	ctx := req.Context()
+
+	// Get trace attributes from semconv
+	attrs := semconv.HTTPClientRequestTraceAttrs(req)
+
+	// Start span
+	spanName := req.Method
+	ctx, span := tracer.Start(ctx,
+		spanName,
+		trace.WithSpanKind(trace.SpanKindClient),
+		trace.WithAttributes(attrs...),
+	)
+
+	// Inject trace context into request headers
+	propagator.Inject(ctx, propagation.HeaderCarrier(req.Header))
+
+	// Update request with new context
+	newReq := req.WithContext(ctx)
+	ictx.SetParam(1, newReq)
```

Comment:
nitpick: same as below, `1` can be const
```suggestion
    const requestParamIndex = 1
	ictx.SetParam(requestParamIndex, newReq)
```